File: tests/matrix/reorder_dims/matrix_reorder_dims_basic.cpp
#include <optitrust.h>
#include <stdlib.h>

int main () {
  int i1 = 0, i2 = 0, i3 = 0;
  const int N1 = 10;
  const int N2 = 10;
  const int N3 = 10;
  int* p = CALLOC3(int, N1, N2, N3);
  p[MINDEX3(N1,N2,N3,i1,i2,i3)];
  return 0;
}

File: tests/matrix/reorder_dims/matrix_reorder_dims_basic.ml
open Optitrust
open Target



let _ = Run.script_cpp (fun _ ->

  !! Matrix_basic.reorder_dims ~order:[2;1;0] [cVarInit "p"];
  !! Matrix_basic.reorder_dims ~rotate_n:2 [cCall "MINDEX3"];

)
